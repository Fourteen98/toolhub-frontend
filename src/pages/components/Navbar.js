import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../assets/ToolhubLogo.svg';

const Navbar = () => (
  <nav className="nav--bar">
    <div className="nav--bar--logo">
      <img className="toolhub--logo" src={logo} alt="toolhub logo" />
      <h1 className="logo--title">ToolHub</h1>
    </div>
    <div className="nav--bar--links">
      <NavLink to="/" className={({ isActive }) => (isActive ? 'active nav--link' : 'inactive nav--link')}>
        Home
      </NavLink>
      <NavLink to="/leaderboard" className={({ isActive }) => (isActive ? 'active nav--link' : 'inactive nav--link')}>
        Leaderboard
      </NavLink>
      <NavLink to="/recordsboard" className={({ isActive }) => (isActive ? 'active nav--link' : 'inactive nav--link')}>
        Records Board
      </NavLink>
    </div>
  </nav>

);

export default Navbar;
