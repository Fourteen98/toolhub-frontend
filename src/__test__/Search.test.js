import React from "react";
import renderer from "react-test-renderer";
import Navbar from "../pages/components/Navbar";

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
