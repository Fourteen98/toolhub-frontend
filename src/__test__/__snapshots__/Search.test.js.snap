// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
Array [
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search for a tool"
      type="text"
      value=""
    />
    <div
      className="button-container"
    >
      <button
        className="home--button"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
      <button
        className="home--button"
        onClick={[Function]}
        type="button"
      >
        Reset
      </button>
    </div>
  </div>,
  <div
    className="search--results"
  >
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Depictor"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Depictor
          </h3>
          <p>
            A tool for creating diagrams and flowcharts
          </p>
          <p>
            https://depictor.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Translate"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Translate
          </h3>
          <p>
            A tool for translating text
          </p>
          <p>
            https://translate.google.com/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="QSML"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            QSML
          </h3>
          <p>
            A tool for creating and editing QSML files
          </p>
          <p>
            https://qsml.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="TinkerCell"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            TinkerCell
          </h3>
          <p>
            A tool for creating and editing TinkerCell files
          </p>
          <p>
            https://tinkercell.com/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Minefield"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Minefield
          </h3>
          <p>
            A tool for creating and editing Minefield files
          </p>
          <p>
            https://minefield.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="BioModels Database"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            BioModels Database
          </h3>
          <p>
            A tool for creating and editing BioModels Database files
          </p>
          <p>
            https://www.ebi.ac.uk/biomodels/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Close to me"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Close to me
          </h3>
          <p>
            A tool for creating and editing Close to me files
          </p>
          <p>
            https://www.closetome.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="MultiSearch"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            MultiSearch
          </h3>
          <p>
            A tool for creating and editing MultiSearch files
          </p>
          <p>
            https://multisearch.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Visualize"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Visualize
          </h3>
          <p>
            A tool for creating and editing Visualize files
          </p>
          <p>
            https://visualize.app/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="BioSimulators"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            BioSimulators
          </h3>
          <p>
            A tool for creating and editing BioSimulators files
          </p>
          <p>
            https://biosimulators.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Wikidata Skim"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Wikidata Skim
          </h3>
          <p>
            A tool for creating and editing Wikidata Skim files
          </p>
          <p>
            https://skim.wmflabs.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Wdqs-proxy"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Wdqs-proxy
          </h3>
          <p>
            A tool for creating and editing Wdqs-proxy files
          </p>
          <p>
            https://wdqs-proxy.toolforge.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Wrdz"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Wrdz
          </h3>
          <p>
            A tool for creating and editing Wrdz files
          </p>
          <p>
            https://wrdz.toolforge.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Wikidata Query Service"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Wikidata Query Service
          </h3>
          <p>
            A tool for creating and editing Wikidata Query Service files
          </p>
          <p>
            https://query.wikidata.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="Molecular Design Toolkit"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            Molecular Design Toolkit
          </h3>
          <p>
            A tool for creating and editing Molecular Design Toolkit files
          </p>
          <p>
            https://molecular-design-toolkit.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="D3"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            D3
          </h3>
          <p>
            A tool for creating and editing D3 files
          </p>
          <p>
            https://d3js.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="D3.js"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            D3.js
          </h3>
          <p>
            A tool for creating and editing D3.js files
          </p>
          <p>
            https://d3js.org/
          </p>
          <p>
            Design
          </p>
        </div>
      </div>
    </div>
    <div
      className="search--results--container"
    >
      <div
        className="search--results--card"
      >
        <div
          className="tool--card--img--container"
        >
          <img
            alt="D5.js"
            className="tool--card--img"
            src="Tool3.png"
          />
        </div>
        <div
          className="search-results-content"
        >
          <h3>
            D5.js
          </h3>
          <p>
            A tool for creating and editing D3.js files
          </p>
          <p />
          <p />
        </div>
      </div>
    </div>
  </div>,
]
`;
